### DISCUSSION TITLE
Enable "Automatically delete head branches"

### DISCUSSION THREAD

**Discussion by 64kramsystem** [fishfight/punchy#146] [Ideas]
On my projects, I always enable the "Automatically delete head branches" Github feature. This automatically gets rid of branches whose PR has been merged.

Since I supposed nobody here recycles branches whose PR have been merged, enabling this feature removes the need to manually cleanup the merged branches (I do use a script actually, but I'd rather have Github do it automatically).

**Comment by odecay** [comment]
haha Ive been making prs from my main branch but I can easily switch to feature branches if we want to go with this :laughing:

**Reply by 64kramsystem** [reply]
Ok. If you don't mind, I'll enable it!

**Comment by zicklag** [comment]
I'm fine with that. I use [`git branchless`](https://github.com/arxanas/git-branchless) locally and don't even have local branches 90% of the time, so that works fine for me. :)

**Comment by 64kramsystem** [comment]
Well, turns out, this option applies to the repository where the PR branch resides (I wasn't expecting to work like that, but it definitely makes sense).

In other words, those who want the PR branches to be automatically deleted (= me only :joy:), just enable the option in the fork.

🤷 :laughing:

**Reply by erlend-sh** [reply]
On that note of GitHub UI, I was gonna point out this sometimes-unknown feature recently, but it might be largely irrelevant now that we’ve added bors:

![4160AEF1-B8C8-469D-9F09-B13FA0EC74B4](https://user-images.githubusercontent.com/583842/181008177-c8142cab-e9e8-4f7f-b30d-6ce0f040102c.jpeg)

**Reply by 64kramsystem** [reply]
I've had a brief discussion about the options with zicklag about this (bors gives different options for merging), and settled for the standard merge commit. I don't know about other users, but I do use bisect, and do look at past commits for reference, and with small commits (assuming they're well written), both tasks are much more effective.
